### DISCUSSION TITLE
programmatic position change not animated

### DISCUSSION THREAD

**Discussion by LinusGeffarth** [LucasMucGH/BottomSheet#75] [Q&A] [answered]
When I programmatically change the position from middle/bottom to top, it works, but it's **not animated**. As animation option, I have set `.spring()`. I've made my own, custom `BottomSheetPosition` if that matters.

`bottomSheetPosition` is `.middle` by default. Then, in a button action, I set `bottomSheetPosition = .top`. This is how I initialize the bottom sheet:

    .bottomSheet(bottomSheetPosition: $bottomSheetPosition, options: [
        .cornerRadius(15),
        .noBottomPosition,
        .animation(Animation.spring()),
        .shadow(color: Color(.sRGB, white: 0, opacity: 0.2), radius: 7, x: 0, y: 0)
        ]) {
        // ...
    } mainContent: {
        // ...
    }

**Comment by lucaszischka** [comment, marked as answer]
Have you tried using
``` swift
withAnimation(.spring()) {
    self.bottomSheetPosition = .top
}
```
when setting the bottom sheet position programmatically?

**Comment by LinusGeffarth** [comment]
that worked, thanks!

may still be an issue though because in the sample code it doesn't look like this extra animation wrapper shouldn't be needed...
